11-node-frameworks/classworks/Agatha Rudko/index.js

```diff
@@ -0,0 +1,62 @@
+const serve = require('koa-static');
+const Koa = require('koa');
+const session = require('koa-generic-session');
+const bodyParser = require('koa-bodyparser');
+const Router = require('@koa/router');
+const joi = require('joi');
+const router = new Router();
+
+const app = new Koa();
+
+
+const schem = joi.object({
+  name: joi.string().required(),
+  surname: joi.string().min(3),
+});
+
+function get(ctx) {
+  const session = { count: ctx.session.count };
+  ctx.body = session.count;
+}
+
+function remove(ctx) {
+  session.count = 0;
+  ctx.body = session.count;
+}
+const increment = (ctx, next) => {
+  ctx.session.count = ctx.session.count || 0;
+  ctx.session.count++;
+  next();
+};
+
+app.use(bodyParser());
+app.use(session());
+app.use(increment);
+app.use(async (ctx) => {
+  switch (ctx.url) {
+    case '/get':
+      await get(ctx);
+      break;
+    case '/remove':
+      await remove(ctx);
+      break;
+  }
+});
+
+router.post('/form', async (ctx, next) => {
+  if (schema.validate(ctx.request.body).error) {
```

Comment:
save the result of result execution `schem.validate(ctx.request.body)` to the variable, you don't need to call it twice because the result will be the same